tailer/fswatcher/linereader.go

```diff
@@ -41,19 +44,32 @@ func (r *lineReader) ReadLine(file io.Reader) (string, bool, error) {
 		err error
 		buf = make([]byte, 512)
 		n   = 0
+		reg = regexp.MustCompile(r.lineDelimiter)
 	)
 	for {
-		newlinePos := bytes.IndexByte(r.remainingBytesFromLastRead, '\n')
-		if newlinePos >= 0 {
+		newlinesLoc := reg.FindAllIndex(r.remainingBytesFromLastRead, 2)
+		var newlinePos int
+		if newlinesLoc != nil {
+			// if first found match is not the first symbol(s)
+			newlinePos = newlinesLoc[0][0]
+			if len(newlinesLoc) == 2 && newlinePos == 0 {
+				// if first found match is message start symbol(s)
+				newlinePos = newlinesLoc[1][0]
+			}
+		}
+		if newlinePos != 0 {
 			l := len(r.remainingBytesFromLastRead)
 			result := make([]byte, newlinePos)
 			copy(result, r.remainingBytesFromLastRead[:newlinePos])
 			copy(r.remainingBytesFromLastRead, r.remainingBytesFromLastRead[newlinePos+1:])
 			r.remainingBytesFromLastRead = r.remainingBytesFromLastRead[:l-(newlinePos+1)]
-			return string(stripWindowsLineEnding(result)), false, nil
+			return stripLineEnding(string(result)), false, nil
 		} else if err != nil {
 			if err == io.EOF {
-				return "", true, nil
+				result := make([]byte, len(r.remainingBytesFromLastRead))
```

Comment:
Is this for cases where the last line in a file does not end with `\n`? It's ok to do this, but if you use the original implementation for cases where line delimiter is `\n`, please remember to add this as well such that `\n` and regular expressions behave the same.